app/models/submission.rb

```diff
@@ -68,11 +68,10 @@ def automated_feedback_pending?
     return false if (representation_exceptioned? || representation_cancelled?) &&
                     (analysis_exceptioned? || analysis_cancelled?)
     return false if representation_not_queued? && analysis_not_queued?
-    return true if !representation_generated? && !analysis_completed?
-
+    return true if !representation_generated? && !analysis_completed? && track.has_representer? && track.has_analyzer?
     return false if has_automated_feedback?
-    return true if representation_queued? || representation_not_queued?
-    return true if analysis_queued? || analysis_not_queued?
+    return true if (representation_queued? || representation_not_queued?) && track.has_representer?
+    return true if (analysis_queued? || analysis_not_queued?) && track.has_analyzer?
```

Comment:
The main downside of using `track.has_representer?` and `track.has_analyzer?` is that if these values ever switch from `false` to `true`, the UI will think an analysis is being done in the background, whereas in fact it won't.

Some possible solutions are:

1. We detect changes to `track.has_analyzer` and `track.has_representer` and automatically queue the corresponding jobs 
2. We store the `has_representer` and `has_analyzer` status on the submission
3. We add an extra `skipped` value to the `representation_status` and `analysis_status` field